src/version.ts

```diff
@@ -0,0 +1,179 @@
+import chalk from 'chalk'
+import fs from 'fs'
+import FileHelpers from './filehelpers'
+// eslint-disable-next-line
+const minimist = require("minimist");
+
+const args = minimist(process.argv.slice(2))
+
+/**
+ * MiraVersion checks the CDK version consistency across the Mira packge and the
+ * applications using Mira.  It is capable of force-patching package.json files
+ * and running the NPM install.
+ */
+export class MiraVersion {
+    static instance: MiraVersion
+    constructor () {
+      if (!MiraVersion.instance) {
+        MiraVersion.instance = this
+      }
+    }
+
+    /**
+     * Checks a particular dependency CDK version.
+     */
+    /* eslint-disable-next-line */
+    checkApplicationDependencyCDKVersion (pkg: any, dep: string, autoFix = true): boolean {
+      if (!dep.startsWith('@aws-cdk') && !dep.startsWith('aws-cdk')) {
+        return false
+      }
+      const cdkVersion = pkg.dependencies[dep]
+      const miraVersion = this.getLocalMiraCDKVersion()
```

Comment:
Could this function be memoized? It seems a bit unnecessary to rerun this function for every entry in the dependency list.

Reply:
Addressed this in latest commit